contracts/priv/notebooksd.yml

```diff
@@ -147,6 +147,181 @@ paths:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Shares'
+        '401':
+          $ref: '#/components/responses/ServerError'
+        '500':
+          $ref: '#/components/responses/ServerError'
+    post:
+      summary: creates a Share
+      operationId: createShare
+      tags:
+        - Share
+      requestBody:
+        $ref: '#/components/requestBodies/ShareParams'
+      responses:
+        '200':
+          description: Share created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Share'
+        '400':
+          $ref: '#/components/responses/ServerError'
+        '401':
+          $ref: '#/components/responses/ServerError'
+        '500':
+          $ref: '#/components/responses/ServerError'
+  '/notebooks/share/{id}':
+    get:
+      summary: get single Share by its ID
+      operationId: getShare
+      tags:
+        - Share
+      parameters:
+        - name: id
+          in: path
+          schema:
+            type: string
+          required: true
+      responses:
+        '200':
+          description: returns the Share record
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Share'
+        '400':
+          $ref: '#/components/responses/ServerError'
+        '401':
+          $ref: '#/components/responses/ServerError'
+        '500':
+          $ref: '#/components/responses/ServerError'
+    delete:
+      summary: deletes a single Share by its ID
+      operationId: deleteShare
+      tags:
+        - Share
+      parameters:
+        - name: id
+          in: path
+          schema:
+            type: string
+          required: true
+      responses:
+        '204':
+          $ref: '#/components/responses/ServerError'
+        '400':
+          $ref: '#/components/responses/ServerError'
+        '401':
+          $ref: '#/components/responses/ServerError'
+        '404':
+          $ref: '#/components/responses/ServerError'
+        '500':
+          $ref: '#/components/responses/ServerError'
+    put:
+      summary: update a single Share by its ID
+      operationId: putShare
+      tags:
+        - Share
+      parameters:
+        - name: id
+          in: path
+          schema:
+            type: string
+          required: true
+      requestBody:
+        $ref: '#/components/requestBodies/ShareParams'
+      responses:
+        '200':
+          description: returns the Share record
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Share'
+        '400':
+          $ref: '#/components/responses/ServerError'
+        '401':
+          $ref: '#/components/responses/ServerError'
+        '500':
+          $ref: '#/components/responses/ServerError'
+  '/api/share/{id}/query/{pipeID}':
```

Comment:
Just wondering why we chose to go without notebooks in the path. Something like this `/api/notebooks/share/.......`?

Reply:
well the original plan was to not limit the scope to notebooks. Could be sharing dashboards or something. But now that we are narrowing the scope to notebooks, you're probably right. I'll probably iterate on that in a future PR tho because it will require changes to several places.